Source/StarRunner2019/HallwayActor.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 Dulce Studios. All Rights Reserved.
+
+#include "HallwayActor.h"
+
+#include "HallwayJointComponent.h"
+#include "HallwayUnitComponent.h"
+
+#include "StarRunner2019Character.h"
+
+#include "Kismet/KismetMathLibrary.h"
+#include "Engine/Classes/Kismet/KismetSystemLibrary.h"
+
+#include <stdlib.h> // for rand function LOL
+
+// Sets default values
+AHallwayActor::AHallwayActor()
+{
+	// Set this actor to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
+	PrimaryActorTick.bCanEverTick = true;
+
+	USceneComponent *RootSceneComponent = CreateDefaultSubobject<USceneComponent>(TEXT("RootScene"));
+	RootComponent = RootSceneComponent;
+
+	Setup();
+}
+
+void AHallwayActor::Setup()
+{
+	UHallwayUnitComponent *InitialHallwayUnitComponent = CreateDefaultSubobject<UHallwayUnitComponent>(TEXT("InitialHallwayUnitComponent"));
+	InitialHallwayUnitComponent->AttachToComponent(GetRootComponent(), FAttachmentTransformRules::KeepRelativeTransform);
+
+	FVector Origin;
+	FVector BoxExtent;
+	float SphereRadius;
+	UKismetSystemLibrary::GetComponentBounds(InitialHallwayUnitComponent, Origin, BoxExtent, SphereRadius);
+
+	float XLength = (BoxExtent.X * 2.0f);
+
+	FRotator Rotator(0.0f, 0.0f, 0.0f);
+	FVector Translation(XLength, 0.0f, 0.0f);
+	FVector Scale(1.0f, 1.0f, 1.0f);
+
+	int HallwayCount = rand() % 5 + 1;
+
+	UE_LOG(LogTemp, Warning, TEXT("Generating %d hallways"), HallwayCount);
+
+	int i;
+	for (i = 1; i < HallwayCount; ++i)
```

Comment:
It's c++11, pls put the int in the for loop unless you're using it elsewhere in the scope.